### PR TITLE
Refactoring Camera as a Component

### PR DESCRIPTION
**Pull request by Rhuagh**
Render pass use Transform of Camera entity to compute the view matrix.

### CODE REVIEW

File: amethyst_renderer/src/pass/flat/interleaved.rs [hunk 1]
```diff
@@ -138,7 +140,8 @@ where
             ..
         } = self;
 
-        let camera = &camera;
+        // TODO: multiple cameras
+        let camera = (&camera, &global).join().next();
```
Comment (torkleyy) [inline amethyst_renderer/src/pass/flat/interleaved.rs]: Maybe log an error in case `next().is_some()`

Comment (torkleyy) [reply amethyst_renderer/src/pass/flat/interleaved.rs]: But then again I'm not sure if we should do this everywhere

**Comment by minecrawler** [comment]
Hey, that's half my PR, however you cleared up one of my major pain points, so that's really awesome :D

1. If possible, could you add the ActiveCamera component we talked about over there, so multiple cameras are possible? Only always take the first ActiveCamera!
2. Shouldn't a camera have both, `Transform` and `LocalTransform`, for example in case you want the camera to follow a mesh? (I hope I got the purpose of `LocalTransform` right this time)

**Comment by Rhuagh** [comment]
For 2. That's up to the user, a LocalTransform will always be converted to a Transform before arriving in the renderer, this is done by Transformsystem.

**Comment by Rhuagh** [comment]
For 1. I'm still debating solutions to this problem

**Comment by omni-viral** [comment]
`ActiveCamera` as optional resource is better. Otherwise changing active camera will have more boilerplate.

**Comment by Rhuagh** [comment]
I think I'm gonna add `ActiveViewport` in a later PR

**Comment by Rhuagh** [comment]
I will add the active camera resource for now, as optional, fallback to first camera if its not set, and fallback to identity from that

File: amethyst_renderer/src/pass/flat/interleaved.rs [hunk 2]
```diff
@@ -140,8 +145,14 @@ where
             ..
         } = self;
 
-        // TODO: multiple cameras
-        let camera = (&camera, &global).join().next();
+        let camera: Option<(&Camera, &Transform)> = active
+            .and_then(|a| {
+                let cam = camera.get(a.entity);
+                let transform = global.get(a.entity);
+                cam.into_iter().zip(transform.into_iter()).next()
```
Comment (omni-viral) [inline amethyst_renderer/src/pass/flat/interleaved.rs]: This is unusual but good alternative to `cam.and_then(|cam| transform.map(|transform| (cam, transform)))` 😄

Comment (Rhuagh) [reply amethyst_renderer/src/pass/flat/interleaved.rs]: That one doesn't work because of borrow rules.

**Pull request merged**